lighthouse-core/runner.js

```diff
@@ -26,8 +26,9 @@ const LHError = require('./lib/lh-error.js');
 
 class Runner {
   /**
-   * @param {(runnerData: {requestedUrl: string, config: Config, driverMock?: Driver}) => Promise<LH.Artifacts>} gatherFn
-   * @param {{config: Config, url?: string, driverMock?: Driver}} runOpts
+   * @template {LH.Config.Config | LH.Config.FRConfig} TConfig
```

Comment:
is this just some code golf?

Reply:
I wouldn't be very good at it if it were ;) 

this is replacing `Config` type requirement with `TConfig` which can be either the legacy or new fraggle rock config, it doesn't care which one so long as they match

Reply:
from how I understand TS they wouldn't have to match, am I missing something that @template does?

(oh, seems you reverted changes to this file, ok)

Reply:
they're not reverted, just part of the filtering changes #11941 that disappeared after I fixed some merge conflicts :)

> from how I understand TS they wouldn't have to match, am I missing something that @template does?

I think so :)

In my understanding `@template {Bar} TFoo` ~= `<TFoo extends Bar>`. Once a invocation parameter constrains the type of TConfig to FRConfig, the other parameter must also be constrained.

[TS Playground Example](https://www.typescriptlang.org/play?useJavaScript=true#code/FAegVGwARlACAXApgWwA4BsCGyoG8BnBAJwEsA7AcygB8pyBXFAIyWIF8oAVAMQHs+0WHDRZiWFPl4DOAMwFD4o8ZLwAKeXwBc3fnwCUUALwA+KADc+pACadmYoSGCyG5AMYJSfclEp8NAgA0UPbEhnjQUFHESAgMxD6hAQbA7MDp4MIIAJ5oSPhqAB46jCxshqYWVrYwTm7eRFCaAHJMrMTGUIXGZvXkBHwYSAB0GHyURfqgEPA5eQXFUERkVBVmljacYHUNCE0CAMokFNRGXT1QfQNDo+OT6cB+agCMwS1t5Y-+AOSa32+HY6rL4vAF8I4rShTJ6-AT-fZ8VplMJAA)